pkg/features/kube_features.go

```diff
@@ -994,7 +995,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	GracefulNodeShutdownBasedOnPodPriority: {Default: true, PreRelease: featuregate.Beta},
 
-	HPAContainerMetrics: {Default: true, PreRelease: featuregate.Beta},
+	HPAContainerMetrics: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.32
```

Comment:
Added `LockToDefault` as I was suggested in another GA graduation PR https://github.com/kubernetes/kubernetes/pull/123481.

Reply:
Good. Note that in future releases the logic using the featuregate must remain in order to support the compatibility versions: https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/4330-compatibility-versions .